lib/screens/last_trip/poller.ex

```diff
@@ -32,14 +33,20 @@ defmodule Screens.LastTrip.Poller do
         :ok = Cache.reset()
         %{state | next_reset: next_reset()}
       else
-        {:ok,
-         %{
-           trip_updates: trip_updates,
-           vehicle_positions: vehicle_positions
-         }} = fetch_trip_updates_and_vehicle_positions()
-
-        update_last_trips(trip_updates)
-        update_recent_departures(trip_updates, vehicle_positions)
+        case fetch_trip_updates_and_vehicle_positions() do
+          {:ok,
+           %{
+             trip_updates: trip_updates,
+             vehicle_positions: vehicle_positions
+           }} ->
+            update_last_trips(trip_updates)
+            update_recent_departures(trip_updates, vehicle_positions)
+
+          {:error, error} ->
+            Logger.warning(
+              "[last_trip_poller] failed while polling for new data: #{inspect(error)}"
```

Comment:
In case there actually is a persistent error, I don't think we should necessarily just keep warning forever, letting this data drift arbitrarily far out of date without notifying us that something's wrong. Perhaps we could incorporate a "failure count" into the process state, and crash when consecutive failures are too high.

Reply:
Seems like a good use case for the `retry` macro. I can add that in no problem.